### PR TITLE
Use mypy --strict

### PR DESCRIPTION
**Pull request by dustalov**
## Description

This pull request enforces a stricter set of mypy type checks by enabling the strict mode. It also fixes several type inconsistencies. As the NumPy type annotations were introduced in version 1.20 (January 2021), some Crowd-Kit installations might broke, but I believe it is a worthy contribution.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation and examples improvement (changes affected documentation and/or examples)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: crowdkit/metrics/data/_classification.py [hunk 1]
```diff
@@ -26,12 +26,18 @@ def _check_answers(answers: pd.DataFrame) -> None:
...
+        return cast(float, row['skill'])
+    else:
+        return (1. - cast(float, row['skill'])) / (n_labels - 1)
 
 
 def _task_consistency(row: pd.Series) -> float:
     """Posterior probability for a single task"""
-    return row[row['aggregated_label']] / row['denominator'] if row['denominator'] != 0 else 0.0
+    if row['denominator'] != 0:
+        return cast(float, row[row['aggregated_label']]) / cast(float, row['denominator'])
```
Comment (DrhF) [inline crowdkit/metrics/data/_classification.py:38]: Why we use `cast`, why not explicitly cast to float?

**Review by DrhF — CHANGES_REQUESTED** [review]

File: crowdkit/metrics/data/_classification.py (hunk 1 above)
Comment (dustalov) [reply crowdkit/metrics/data/_classification.py:38]: My first idea was ‘why not’, but I think you're more correct.

File: crowdkit/metrics/data/_classification.py [hunk 2]
```diff
@@ -245,4 +251,4 @@ def alpha_krippendorff(answers: pd.DataFrame,
     """
     _check_answers(answers)
     data: List[Tuple[Any, Hashable, Hashable]] = answers[['worker', 'task', 'label']].values.tolist()
-    return AnnotationTask(data, distance).alpha()
+    return cast(float, AnnotationTask(data, distance).alpha())
```
Comment (DrhF) [inline crowdkit/metrics/data/_classification.py:254]:
```suggestion
    return float(AnnotationTask(data, distance).alpha())
```

**Review by DrhF — CHANGES_REQUESTED** [review]
Some more suggestions about `cast` usage. Sorry for not mentioning them on the previous review

**Review by dustalov — COMMENTED** [review]
More casts to float

**Comment by dustalov** [comment]
I spent some considerable amount of time in the Web interface of GitHub to achieve nothing.

**Review by DrhF — APPROVED** [review]
Now it looks good:-)

**Pull request merged**